app/src/test/java/info/nightscout/androidaps/plugins/general/autotune/AutotunePrepTest.kt

```diff
@@ -87,10 +89,93 @@ class AutotunePrepTest : TestBaseWithProfile() {
     }
 
 
+    @Test
+    fun autotunePrepTest2() { // Test if categorisation without carbs (full UAM) and categorize UAM as basal false is Ok
+        val inputIobJson = File("src/test/res/autotune/test2/oaps-iobCalc.2022-05-21.json").readText() //json files build with iob/activity calculated by OAPS
+        val iobOapsCalcul = buildIobOaps(JSONArray(inputIobJson))
+        autotuneIob = TestAutotuneIob(aapsLogger, repository, profileFunction, sp, dateUtil, activePlugin, autotuneFS, iobOapsCalcul)
+        autotunePrep = AutotunePrep(sp, dateUtil, autotuneFS, autotuneIob)
+        val inputProfileJson = File("src/test/res/autotune/test2/profile.pump.json").readText()
+        val inputProfile = atProfileFromOapsJson(JSONObject(inputProfileJson), dateUtil)!!
+        val prepjson = File("src/test/res/autotune/test2/autotune.2022-05-21.json").readText()
+        val oapsPreppedGlucose = PreppedGlucose(JSONObject(prepjson), dateUtil) //prep data calculated by OpenAPS autotune
+        val oapsEntriesJson = File("src/test/res/autotune/test2/aaps-entries.2022-05-21.json").readText()
+        autotuneIob.glucose =  buildGlucose(JSONArray(oapsEntriesJson))
+        val oapsTreatmentsJson = File("src/test/res/autotune/test2/aaps-treatments.2022-05-21.json").readText()
+        autotuneIob.meals =  buildMeals(JSONArray(oapsTreatmentsJson))  //Only meals is used in unit test, Insulin only used for iob calculation
+        autotuneIob.boluses = buildBoluses(oapsPreppedGlucose) //Values from oapsPrepData because linked to iob calculation method for TBR
+        `when`(sp.getDouble(R.string.key_openapsama_min_5m_carbimpact, 3.0)).thenReturn(min5mCarbImpact)
+        `when`(sp.getBoolean(R.string.key_autotune_categorize_uam_as_basal, false)).thenReturn(false)           // CategorizeUAM as Basal = False
+        val aapsPreppedGlucose = autotunePrep.categorizeBGDatums(inputProfile, inputProfile.localInsulin, false)
+        try {
+            aapsPreppedGlucose?.let {       // compare all categorization calculated by aaps plugin (aapsPreppedGlucose) with categorization calculated by OpenAPS (oapsPreppedGlucose)
+                for (i in aapsPreppedGlucose.crData.indices)
+                    Assert.assertTrue(oapsPreppedGlucose.crData[i].equals(aapsPreppedGlucose.crData[i]))
+                for (i in aapsPreppedGlucose.csfGlucoseData.indices)
+                    Assert.assertTrue(oapsPreppedGlucose.csfGlucoseData[i].equals(aapsPreppedGlucose.csfGlucoseData[i]))
+                aapsLogger.debug("Oaps ISF Size ${oapsPreppedGlucose.isfGlucoseData.size} / Aaps ISF Size ${aapsPreppedGlucose.isfGlucoseData.size}")
```

Comment:
I already tried to use aapsLogger here, but it didn't help me (I can't see log information in CircleCI).
All these tests are Ok on my computer, and I can't reproduce what happens during automated dotests...

```
<============-> 99% EXECUTING [49s]> :app:testFullDebugUnitTest > Executing test info.nightscout.androidaps.interf> :app:testFullDebugUnitTest > Executing test info.nightscout...general.autotun> :app:testFullDebugUnitTest > 12 tests completed> :app:testFullDebugUnitTest > Executing test info.nightscout.androidaps.plugin<============-> 99% EXECUTING [50s]
info.nightscout.androidaps.plugins.general.autotune.AutotunePrepTest > autotunePrepTest2 FAILED
    java.lang.AssertionError: expected:<114> but was:<115>
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at info.nightscout.androidaps.plugins.general.autotune.AutotunePrepTest.autotunePrepTest2(AutotunePrepTest.kt:117)
```